keras_cv/layers/preprocessing/rand_augment.py

```diff
@@ -0,0 +1,277 @@
+        seed=None,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.num_layers = num_layers
+        self.magnitude = magnitude
+        self.value_range = value_range
+        if magnitude < 0.0 or magnitude > 10.0:
+            raise ValueError(
+                f"`magnitude` must be in the range [0, 10], got `magnitude={magnitude}`"
+            )
+        self.magnitude_standard_deviation = magnitude_standard_deviation
+        self.probability_to_apply = probability_to_apply
+
+        policy = create_rand_augment_policy(magnitude, magnitude_standard_deviation)
+
+        self.auto_contrast = cv_preprocessing.AutoContrast(
+            **policy["auto_contrast"], value_range=(0, 255), seed=seed
+        )
+        self.equalize = cv_preprocessing.Equalization(
+            **policy["equalize"], value_range=(0, 255), seed=seed
+        )
+
+        self.solarize = cv_preprocessing.Solarization(
+            **policy["solarize"], value_range=(0, 255), seed=seed
+        )
+        self.solarize_add = cv_preprocessing.Solarization(
+            **policy["solarize_add"], value_range=(0, 255), seed=seed
+        )
+        self.invert = cv_preprocessing.Solarization(
+            **policy["invert"], value_range=(0, 255), seed=seed
+        )
+
+        self.color = cv_preprocessing.RandomColorDegeneration(
+            **policy["color"], seed=seed
+        )
+        self.contrast = cv_preprocessing.RandomContrast(**policy["contrast"], seed=seed)
+        self.brightness = cv_preprocessing.RandomBrightness(
+            **policy["brightness"], seed=seed
+        )
+        self.shear_x = cv_preprocessing.RandomShear(**policy["shear_x"], seed=seed)
+        self.shear_y = cv_preprocessing.RandomShear(**policy["shear_y"], seed=seed)
+        self.translate_x = cv_preprocessing.RandomTranslation(
+            **policy["translate_x"], seed=seed
+        )
+        self.translate_y = cv_preprocessing.RandomTranslation(
+            **policy["translate_y"], seed=seed
+        )
+        self.cutout = cv_preprocessing.RandomCutout(**policy["cutout"], seed=seed)
+
+        self.augmentation_layers = [
+            self.auto_contrast,
+            self.equalize,
+            self.solarize,
+            self.solarize_add,
+            self.invert,
+            self.color,
+            self.contrast,
+            self.brightness,
+            self.shear_x,
+            self.shear_y,
+            self.translate_x,
+            self.translate_y,
+            self.cutout,
+        ]
+        self.auto_vectorize = False
+
+    def _augment(self, sample):
+        sample["images"] = preprocessing_utils.transform_value_range(
+            sample["images"], self.value_range, (0, 255)
+        )
+        augmented_sample = sample
+        for _ in range(self.num_layers):
+            selected_op = tf.random.uniform(
+                (), maxval=len(self.augmentation_layers) + 1, dtype=tf.int32
+            )
```

Comment:
we should use seed whenever tf.random ops appears.

Reply:
Should we use stateless ops and a Generator?

(This should likely be handled at the level of the base augmentation layer -- maybe it already is)

Reply:
Done, used the generator.